### PR TITLE
U4X-436:Fetching data for a given patient to the care panel summary

### PR DESCRIPTION
**Pull request by Daphne210**
https://github.com/METS-Programme/esm-ugandaemr-core/assets/89844768/7ff1bb36-7101-4b0a-a24f-9ef5e6750256

https://metsprogramme.atlassian.net/browse/U4X-436

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "editor.formatOnSave": true,
```
Comment (jabahum) [inline .vscode/settings.json:2]: please remove this

File: packages/esm-care-panel-app/src/care-panel/care-panel.component.tsx [hunk 2]
```diff
@@ -56,32 +51,38 @@ const CarePanel: React.FC<CarePanelProps> = ({ patientUuid, formEntrySub, launch
...
-          <ProgramEnrollment
-            patientUuid={patientUuid}
-            programName={switcherHeaders[switchItem?.index]}
-            enrollments={patientEnrollments}
-            formEntrySub={formEntrySub}
-            launchPatientWorkspace={launchPatientWorkspace}
-          />
-
-          <CarePrograms patientUuid={patientUuid} />
+          {switchItem?.name === 'HIV Program' ? (
```
Comment (jabahum) [inline packages/esm-care-panel-app/src/care-panel/care-panel.component.tsx:67]: move this to something like this ```  export const  program = {HIV:"HIV Program"}```

File: packages/esm-care-panel-app/src/program-enrollment/program-enrollment.resource.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          resultsByConcept[observedConceptCode] = [];
+        }
+        // Push the value into the correct array based on conceptCode
+        resultsByConcept[observedConceptCode].push(value);
+      }
+    });
+
+    return resultsByConcept;
+  }, [data, conceptUuids]);
+  // console.info(JSON.stringify(data, null, 2));
```
Comment (jabahum) [inline packages/esm-care-panel-app/src/program-enrollment/program-enrollment.resource.ts:45]: lets remove this unnecessary code

**Review by jabahum — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (Daphne210) [reply .vscode/settings.json:2]: This has been resolved

File: packages/esm-care-panel-app/src/care-panel/care-panel.component.tsx (hunk 2 above)
Comment (Daphne210) [reply packages/esm-care-panel-app/src/care-panel/care-panel.component.tsx]: @jabahum  This has been resolved

File: packages/esm-care-panel-app/src/program-enrollment/program-enrollment.resource.ts (hunk 3 above)
Comment (Daphne210) [reply packages/esm-care-panel-app/src/program-enrollment/program-enrollment.resource.ts]: @jabahum This has been resolved

File: packages/esm-care-panel-app/src/care-panel/care-panel.component.tsx [hunk 4]
```diff
@@ -56,32 +52,38 @@ const CarePanel: React.FC<CarePanelProps> = ({ patientUuid, formEntrySub, launch
...
-            patientUuid={patientUuid}
-            programName={switcherHeaders[switchItem?.index]}
-            enrollments={patientEnrollments}
-            formEntrySub={formEntrySub}
-            launchPatientWorkspace={launchPatientWorkspace}
-          />
-
-          <CarePrograms patientUuid={patientUuid} />
+          {programs.HIV ? (
+            <>
```
Comment (jabahum) [inline packages/esm-care-panel-app/src/care-panel/care-panel.component.tsx:69]: we can remove this extract element tag <></>

**Review by Mwanje — APPROVED** [review]

**Review by jabahum — APPROVED** [review]

**Pull request merged**